ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -359,6 +348,32 @@ private void analyzeReplLoad(ASTNode ast) throws SemanticException {
     }
   }
 
+  private void setConfigs(ASTNode node) throws SemanticException {
+    Map<String, String> replConfigs = DDLSemanticAnalyzer.getProps(node);
+    if (null != replConfigs) {
+      for (Map.Entry<String, String> config : replConfigs.entrySet()) {
+        String key = config.getKey();
+        // don't set the query id in the config
+        if (key.equalsIgnoreCase(HIVEQUERYID.varname)) {
+          String queryTag = config.getValue();
+          if (!StringUtils.isEmpty(queryTag)) {
+            QueryState.setMapReduceJobTag(conf, queryTag);
```

Comment:
Do we need to add the tag to queryState.queryConf as well?

Reply:
it is already added when query state is created